### PR TITLE
yarn.lock fix for semantic-release/commit-analyzer

### LINKED ISSUE #191: incorrect version of semantic-release/commit-analyzer in the lockfile
Opening a new issue to avoid cluttering the other one.

As mentioned in #190, currently running `yarn install` fails for me, on so-very-many different versions of node, npm, yarn, etc, and on 2 different Ubuntu machines.

error message:
`error An unexpected error occurred: "https://registry.yarnpkg.com/@semantic-release/commit-analyzer/-/commit-analyzer-7.0.0-beta-.2.tgz: Request failed \"404 Not Found\"".`

After looking into it a fair bit, also contacting npm support, it seems like the version referenced in `yarn.lock` is no longer there, perhaps because it was named incorrectly (dash before a dot in `beta-.2`). 

I believe for many people it might work - if they have that package in their cache.

The simplest way for me to finally get it to work was, after cleaning the cache and node modules (so basically starting from scratch), running:
`yarn upgrade semantic-release/commit-analyzer` 

Will PR the resulting lockfile in a minute.

Unfortunately some tests fail, but omniboard seems to work... also I'm not sure if the failing tests are related

### PR DESCRIPTION
**Pull request by tpietruszka**
Closes #191 (hopefully)

Some tests might be failing - but not sure if it is related.

### CODE REVIEW

**Comment by vivekratnavel** [comment]
Looks like commit-analyzer@7.0.0 is incompatible with node v8.

```
error @semantic-release/commit-analyzer@7.0.0: The engine "node" is incompatible with this module. Expected version ">=10.13". Got "8.17.0"
```

Can you remove node 8 from travis CI

**Comment by vivekratnavel** [comment]
https://github.com/vivekratnavel/omniboard/blob/master/.travis.yml#L7

**Comment by tpietruszka** [comment]
Done, thanks, I should have checked. 

Or did you intend to keep the v8 compatibility? There should be a way...

BTW, for some reason, 3 tests from 1 test suite (/RunsTable/runsTable.test.js) are still failing for me locally, but I see that they are passing during CI. Any ideas where such differences can be coming from?

**Comment by vivekratnavel** [comment]
@tpietruszka Thanks for the PR! Which tests are failing in local? And which version of node are you running?

**Comment by tpietruszka** [comment]
@vivekratnavel thank you very much for all your work, unfortunately I still cannot make the tests pass.

I've tried multiple versions of node, v10 and recently v12

I will describe my last attempt in detail. 

Last commit: ab9cc4487d43372b98991cb029a876cc8e285997 (2.7.2)

Node: v10.19.0, yarn 1.15.2 (also tried with the newest one, I got 1.15.2 from one of the Travis logs)

1. cleaned caches, removed node_modules and web/node_modules
2. `yarn --frozen-lockfile` (also tried normal `yarn install` previously, ended up editing the lockfile AFAIR
3. `cd web; CI=true yarn run test > test.log 2>&1` . Ran like this to save logs,

<details><summary>Here are the logs:</summary>
<pre>
yarn run v1.15.2
$ TZ='America/New_York' node --expose-gc --max-old-space-size=2048 scripts/test.js --env=jsdom --runInBand --logHeapUsage --detectOpenHandles --forceExit
FAIL src/components/RunsTable/runsTable.test.js (42.256s, 426 MB heap size)
  ● Console

    console.error node_modules/jsdom/lib/jsdom/virtual-console.js:29
      Error: Uncaught [TypeError: Cannot read property 'length' of undefined]
          at reportException (/mnt/Data/projects/omniboard/web/node_modules/jsdom/lib/jsdom/living/helpers/runtime-script-errors.js:62:24)
          at Timeout.callback [as _onTimeout] (/mnt/Data/projects/omniboard/web/node_modules/jsdom/lib/jsdom/browser/Window.js:645:7)
          at ontimeout (timers.js:436:11)
          at tryOnTimeout (timers.js:300:5)
          at listOnTimeout (timers.js:263:5)
          at Timer.processTimers (timers.js:223:10) TypeError: Cannot read property 'length' of undefined
          at RunsTable._buildRunsQuery (/mnt/Data/projects/omniboard/web/src/components/RunsTable/runsTable.js:342:29)
          at RunsTable.loadPartialUpdates (/mnt/Data/projects/omniboard/web/src/components/RunsTable/runsTable.js:683:33)
          at RunsTable (/mnt/Data/projects/omniboard/web/src/components/RunsTable/runsTable.js:199:10)
          at Timeout.callback [as _onTimeout] (/mnt/Data/projects/omniboard/web/node_modules/jsdom/lib/jsdom/browser/Window.js:643:19)
          at ontimeout (timers.js:436:11)
          at tryOnTimeout (timers.js:300:5)
          at listOnTimeout (timers.js:263:5)
          at Timer.processTimers (timers.js:223:10)

  ● RunsTable › should add or remove filters › should load value dropdown options for status

    No request to respond to!

      100 | 
      101 |     mockAxios.mockResponse({status: 200, data: runsResponse});
    > 102 |     mockAxios.mockResponse({status: 200, data: tagsResponse});
          |               ^
      103 |     mockAxios.mockResponse({status: 200, data: countResponse});
      104 |     await tick();
      105 |   };

      at Function.mockResponse (node_modules/jest-mock-axios/lib/mock-axios.ts:123:11)
      at initialRequestResponse (src/components/RunsTable/runsTable.test.js:102:15)

  ● RunsTable › should add or remove filters › should change value dropdown to multiselect

    TypeError: Cannot read property 'length' of undefined

      340 |       }
      341 | 
    > 342 |       if (customColumnsData.length > 0) {
          |                             ^
      343 |         // Select those paths that are not already present in the select to avoid two conflicting paths error
      344 |         const customColumnPaths = customColumnsData.filter(column => !select.split(',')
      345 |           .some(col => col === (column.config_path && column.config_path.split('.')[0])));

      at RunsTable._buildRunsQuery (src/components/RunsTable/runsTable.js:342:29)
      at RunsTable.loadPartialUpdates (src/components/RunsTable/runsTable.js:683:33)
      at RunsTable (src/components/RunsTable/runsTable.js:199:10)
      at Timeout.callback [as _onTimeout] (node_modules/jsdom/lib/jsdom/browser/Window.js:643:19)

  ● RunsTable › should add or remove filters › value dropdown should allow multiple values

    No request to respond to!

      960 |       expect(wrapper.update().find('[test-attr="filter-column-value"]').at(1).props().isMulti).toBeFalsy();
      961 | 
    > 962 |       mockAxios.mockResponse({status: 200, data: ['host1', 'host2']});
          |                 ^
      963 |       wrapper.find('[test-attr="filter-column-operator-dropdown"]').at(1).prop('onChange')({value: '$in'});
      964 | 
      965 |       expect(wrapper.update().find('[test-attr="filter-column-value"]').at(1).props().isMulti).toBeTruthy();

      at Function.mockResponse (node_modules/jest-mock-axios/lib/mock-axios.ts:123:11)
      at Object.it (src/components/RunsTable/runsTable.test.js:962:17)

  ● RunsTable › should add or remove filters › value dropdown should allow multiple values

    No request to respond to!

       99 |     await tick();
      100 | 
    > 101 |     mockAxios.mockResponse({status: 200, data: runsResponse});
          |               ^
      102 |     mockAxios.mockResponse({status: 200, data: tagsResponse});
      103 |     mockAxios.mockResponse({status: 200, data: countResponse});
      104 |     await tick();

      at Function.mockResponse (node_modules/jest-mock-axios/lib/mock-axios.ts:123:11)
      at initialRequestResponse (src/components/RunsTable/runsTable.test.js:101:15)

  ● RunsTable › should fetch next set of runs › and handle success

    No request to respond to!

      100 | 
      101 |     mockAxios.mockResponse({status: 200, data: runsResponse});
    > 102 |     mockAxios.mockResponse({status: 200, data: tagsResponse});
          |               ^
      103 |     mockAxios.mockResponse({status: 200, data: countResponse});
      104 |     await tick();
      105 |   };

      at Function.mockResponse (node_modules/jest-mock-axios/lib/mock-axios.ts:123:11)
      at initialRequestResponse (src/components/RunsTable/runsTable.test.js:102:15)

  ● RunsTable › should fetch next set of runs › and handle success

    expect(received).toHaveLength(expected)

    Expected length: 2
    Received length: 6
    Received array:  [["/api/v1/Runs/count", {"params": {"limit": 4, "order_by": "-1", "query": "{}", "select": "_id,experiment.name,host.hostname,format,command,start_time,status,omniboard.tags,omniboard.notes,resources,heartbeat,duration,result,stop_time,pretrain_loss_min,config.message,config.recipient,config.seed,config.train,config.tags,meta.comment", "sort_by": "_id"}}], ["/api/v1/Omniboard.Metric.Columns"], ["/api/v1/Omniboard.Custom.Columns"], ["/api/v1/Runs", {"params": {"limit": 4, "order_by": "-1", "query": "{}", "select": "_id,experiment.name,host.hostname,format,command,start_time,status,omniboard.tags,omniboard.notes,resources,heartbeat,duration,result,stop_time,pretrain_loss_min,config.message,config.recipient,config.seed,config.train,config.tags,meta.comment", "sort_by": "_id"}}], ["/api/v1/Runs", {"params": {"distinct": "omniboard.tags"}}], ["/api/v1/Runs/count", {"params": {"query": "{}"}}]]

      1023 |       const countResponse = {count: 5};
      1024 | 
    > 1025 |       expect(mockAxios.get.mock.calls).toHaveLength(2);
           |                                        ^
      1026 |       expect(wrapper.state().dataVersion).toEqual(0);
      1027 | 
      1028 |       mockAxios.mockResponse({status: 200, data: runsResponse});

      at Object.it (src/components/RunsTable/runsTable.test.js:1025:40)

  ● RunsTable › should handle updateRuns › for deletion of columns

    No request to respond to!

      1098 | 
      1099 |       mockAxios.mockResponse({status: 200, data: runsResponse});
    > 1100 |       mockAxios.mockResponse({status: 200, data: tagsResponse});
           |                 ^
      1101 |       mockAxios.mockResponse({status: 200, data: countResponse});
      1102 |       await tick();
      1103 | 

      at Function.mockResponse (node_modules/jest-mock-axios/lib/mock-axios.ts:123:11)
      at Object.it (src/components/RunsTable/runsTable.test.js:1100:17)

PASS src/components/MetricsPlotView/metricsPlotView.test.js (201 MB heap size)
PASS src/components/DrillDownView/drillDownView.test.js (191 MB heap size)
PASS src/components/CompareRunsModal/compareRunsModal.test.js (192 MB heap size)
PASS src/components/SourceFilesView/sourceFilesView.test.js (184 MB heap size)
PASS src/components/ConfigCompareView/configCompareView.test.js (184 MB heap size)
PASS src/components/App/index.test.js (194 MB heap size)
PASS src/components/CapturedOutCompareView/capturedOutCompareView.test.js (189 MB heap size)
PASS src/components/SettingsModal/settingsModal.test.js (185 MB heap size)
PASS src/components/SourceFilesCompareView/sourceFilesCompareView.test.js (182 MB heap size)
PASS src/components/CustomColumnModal/customColumnModal.test.js (180 MB heap size)
PASS src/components/MetricColumnModal/metricColumnModal.test.js (180 MB heap size)
PASS src/components/Helpers/cells.test.js (183 MB heap size)
PASS src/components/MetricsCompareView/metricsCompareView.test.js (209 MB heap size)
PASS src/routes.test.js (216 MB heap size)
PASS src/components/FilePreview/filePreview.test.js (206 MB heap size)
PASS src/components/CapturedOutView/capturedOutView.test.js (205 MB heap size)
PASS src/components/XEditable/editableTextArea.test.js (177 MB heap size)
PASS src/components/Helpers/dataListWrapper.test.js (176 MB heap size)
PASS src/components/Helpers/utils.test.js (174 MB heap size)
PASS src/components/NotFound/index.test.js (174 MB heap size)
PASS src/components/XEditable/xEditable.test.js (174 MB heap size)

Summary of all failing tests
FAIL src/components/RunsTable/runsTable.test.js (42.256s, 426 MB heap size)
  ● RunsTable › should add or remove filters › should load value dropdown options for status

    No request to respond to!

      100 | 
      101 |     mockAxios.mockResponse({status: 200, data: runsResponse});
    > 102 |     mockAxios.mockResponse({status: 200, data: tagsResponse});
          |               ^
      103 |     mockAxios.mockResponse({status: 200, data: countResponse});
      104 |     await tick();
      105 |   };

      at Function.mockResponse (node_modules/jest-mock-axios/lib/mock-axios.ts:123:11)
      at initialRequestResponse (src/components/RunsTable/runsTable.test.js:102:15)

  ● RunsTable › should add or remove filters › should change value dropdown to multiselect

    TypeError: Cannot read property 'length' of undefined

      340 |       }
      341 | 
    > 342 |       if (customColumnsData.length > 0) {
          |                             ^
      343 |         // Select those paths that are not already present in the select to avoid two conflicting paths error
      344 |         const customColumnPaths = customColumnsData.filter(column => !select.split(',')
      345 |           .some(col => col === (column.config_path && column.config_path.split('.')[0])));

      at RunsTable._buildRunsQuery (src/components/RunsTable/runsTable.js:342:29)
      at RunsTable.loadPartialUpdates (src/components/RunsTable/runsTable.js:683:33)
      at RunsTable (src/components/RunsTable/runsTable.js:199:10)
      at Timeout.callback [as _onTimeout] (node_modules/jsdom/lib/jsdom/browser/Window.js:643:19)

  ● RunsTable › should add or remove filters › value dropdown should allow multiple values

    No request to respond to!

      960 |       expect(wrapper.update().find('[test-attr="filter-column-value"]').at(1).props().isMulti).toBeFalsy();
      961 | 
    > 962 |       mockAxios.mockResponse({status: 200, data: ['host1', 'host2']});
          |                 ^
      963 |       wrapper.find('[test-attr="filter-column-operator-dropdown"]').at(1).prop('onChange')({value: '$in'});
      964 | 
      965 |       expect(wrapper.update().find('[test-attr="filter-column-value"]').at(1).props().isMulti).toBeTruthy();

      at Function.mockResponse (node_modules/jest-mock-axios/lib/mock-axios.ts:123:11)
      at Object.it (src/components/RunsTable/runsTable.test.js:962:17)

  ● RunsTable › should add or remove filters › value dropdown should allow multiple values

    No request to respond to!

       99 |     await tick();
      100 | 
    > 101 |     mockAxios.mockResponse({status: 200, data: runsResponse});
          |               ^
      102 |     mockAxios.mockResponse({status: 200, data: tagsResponse});
      103 |     mockAxios.mockResponse({status: 200, data: countResponse});
      104 |     await tick();

      at Function.mockResponse (node_modules/jest-mock-axios/lib/mock-axios.ts:123:11)
      at initialRequestResponse (src/components/RunsTable/runsTable.test.js:101:15)

  ● RunsTable › should fetch next set of runs › and handle success

    No request to respond to!

      100 | 
      101 |     mockAxios.mockResponse({status: 200, data: runsResponse});
    > 102 |     mockAxios.mockResponse({status: 200, data: tagsResponse});
          |               ^
      103 |     mockAxios.mockResponse({status: 200, data: countResponse});
      104 |     await tick();
      105 |   };

      at Function.mockResponse (node_modules/jest-mock-axios/lib/mock-axios.ts:123:11)
      at initialRequestResponse (src/components/RunsTable/runsTable.test.js:102:15)

  ● RunsTable › should fetch next set of runs › and handle success

    expect(received).toHaveLength(expected)

    Expected length: 2
    Received length: 6
    Received array:  [["/api/v1/Runs/count", {"params": {"limit": 4, "order_by": "-1", "query": "{}", "select": "_id,experiment.name,host.hostname,format,command,start_time,status,omniboard.tags,omniboard.notes,resources,heartbeat,duration,result,stop_time,pretrain_loss_min,config.message,config.recipient,config.seed,config.train,config.tags,meta.comment", "sort_by": "_id"}}], ["/api/v1/Omniboard.Metric.Columns"], ["/api/v1/Omniboard.Custom.Columns"], ["/api/v1/Runs", {"params": {"limit": 4, "order_by": "-1", "query": "{}", "select": "_id,experiment.name,host.hostname,format,command,start_time,status,omniboard.tags,omniboard.notes,resources,heartbeat,duration,result,stop_time,pretrain_loss_min,config.message,config.recipient,config.seed,config.train,config.tags,meta.comment", "sort_by": "_id"}}], ["/api/v1/Runs", {"params": {"distinct": "omniboard.tags"}}], ["/api/v1/Runs/count", {"params": {"query": "{}"}}]]

      1023 |       const countResponse = {count: 5};
      1024 | 
    > 1025 |       expect(mockAxios.get.mock.calls).toHaveLength(2);
           |                                        ^
      1026 |       expect(wrapper.state().dataVersion).toEqual(0);
      1027 | 
      1028 |       mockAxios.mockResponse({status: 200, data: runsResponse});

      at Object.it (src/components/RunsTable/runsTable.test.js:1025:40)

  ● RunsTable › should handle updateRuns › for deletion of columns

    No request to respond to!

      1098 | 
      1099 |       mockAxios.mockResponse({status: 200, data: runsResponse});
    > 1100 |       mockAxios.mockResponse({status: 200, data: tagsResponse});
           |                 ^
      1101 |       mockAxios.mockResponse({status: 200, data: countResponse});
      1102 |       await tick();
      1103 | 

      at Function.mockResponse (node_modules/jest-mock-axios/lib/mock-axios.ts:123:11)
      at Object.it (src/components/RunsTable/runsTable.test.js:1100:17)


Test Suites: 1 failed, 21 passed, 22 total
Tests:       5 failed, 205 passed, 210 total
Snapshots:   36 passed, 36 total
Time:        66.987s
Ran all test suites.
error Command failed with exit code 1.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
</pre>
</details>

**Comment by vivekratnavel** [comment]
This is strange since I have never seen such errors in Travis. It would be great if I could reproduce these errors. Could you please share your OS version? I will try it out in a VM. Thanks!

**Comment by tpietruszka** [comment]
Both cases Ubuntu 18.04.

Actually, I've just tried on a 3rd system, unfortunately also Ubuntu 18.04. 
It had no node installed, so I :
- installed nvm, 
- `nvm install 12`,  (+re-login)
- cloned the repository, 
- `yarn install` (lockfile got modified a bit)
- `cd web; yarn run tests`

This time 15 tests from the runsTable suite failed. 

The machine running it was weaker than the previously used machines, and more tests failed. All in the test suite which takes a long time to execute. Many errors mention kind of a missing request...

do you think some requests/functions/... might be hitting a timeout?

**Comment by tpietruszka** [comment]
Yeah...

All tests pass on my laptop if I disable power saving. 
If I enable it, several tests fail.
(power saving limits single-core performance)

So basically all this mess was about timeouts set somewhere...

**Comment by vivekratnavel** [comment]
Ah.. Now, it makes sense. Timeouts are probable because the API calls are all async and most of the tests use `await tick()` to wait until the response is processed. 

https://github.com/vivekratnavel/omniboard/blob/b18a2a2a3ea4103b1b0a8da2c582982d161e2f4c/web/config/jest/jestSetup.js#L19

You can try tweaking the setTimeout [here](https://github.com/vivekratnavel/omniboard/blob/b18a2a2a3ea4103b1b0a8da2c582982d161e2f4c/web/config/jest/jestSetup.js#L19) by setting a value > 0 and see if it helps reduce the number of test failures.

**Comment by tpietruszka** [comment]
Setting a non-zero value here breaks even more tests (other timeouts are hit) and slows the tests down significantly.

I've tried setting `jest.setTimeout()` in some places, with no success. Still, if the system is busy, throttled or slow - some tests fail.

If I understand correctly (based on the logs above)  the problems appears very early in the tests -, 
in `initialRequestResponse` there is an exception of "no request to respond to".
Seems related to: https://github.com/knee-cola/jest-mock-axios/issues/28

Looks like there is a possible execution order when parts of `initialRequestResponse` (called from individual failing tests) get executed before the corresponding axios calls.
https://github.com/vivekratnavel/omniboard/blob/b18a2a2a3ea4103b1b0a8da2c582982d161e2f4c/web/src/components/RunsTable/runsTable.test.js#L96-L106

Possibly related:  axios calls are made inside `runsTable.loadData` which is called from `componentDidMount()` after a non-zero timeout (1ms, but still).

At the same time, if I understand correctly, `await tick()` only waits for the next macro-task, which might be earlier, if not too many micro-tasks got pushed into the queue? (but I did try setting the 1ms timeout to 0 - nothing changed)

Another observation: the errors seem to be raised always(?) in lines 101, 102, 1100, 1101 - so about tagsResponse or runsResponse, not about the (earlier) metricColumns and customColumns.


To sum up: while I know this is not very urgent,  because the tests mostly work, I no longer believe this is about particular amount of work not being done in a particular time, which would be solved by increasing a time limit. It looks to me more like race condition, which is generally not something worth having in any codebase...

Apologies for my low Node-related knowledge, I do not have much experience with it (e.g. only read about the internals of the event loop today), so possibly some of my analysis makes no sense...

Still, I think it is a problem worth solving at some point because it might appear again unpredictably after some unrelated changes.

**Comment by vivekratnavel** [comment]
@tpietruszka Sorry for the delayed response and thanks for the detailed analysis on this issue. 

Here are my thoughts:
Your observation about errors always getting raised in lines 101, 102 is valid and I can see why it might happen. It is because, request for metricColumns and customColumns are triggered together when component mounts and the remaining requests are only triggered after processing the response of first two requests and building query for the next set of requests. It is possible that in slower machines, the second set of requests are not triggered yet when the test executes `mockAxios.mockResponse({status: 200, data: runsResponse})` and ends up throwing an exception. 

I think that solving this race condition issue will take a lot of effort and we might not be able to solve without tweaking mockAxios library. But, I am happy that I have never encountered this issue with CI or in any of my dev machines to hinder my progress. So, I have decided not to spend any time on fixing this issue any sooner. But, you are more than welcome to take a shot at it or propose any suggestions to fix this. Thanks!
